### PR TITLE
expression: make inferring of decimal for unix_timestamp be more consistent with MySQL

### LINKED ISSUE #6221: `unix_timestamp`'s return type is not compatible
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
mysql> select unix_timestamp('2016-01-01');
+------------------------------+
| unix_timestamp('2016-01-01') |
+------------------------------+
|            1451577600.000000 |
+------------------------------+
1 row in set (0.00 sec)

2. What did you expect to see?

mysql> select unix_timestamp('2016-01-01');
+------------------------------+
| unix_timestamp('2016-01-01') |
+------------------------------+
|                   1451577600 |
+------------------------------+
1 row in set (0.00 sec)

3. What did you see instead?

Should be the same, but it's different.

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

57afbe2680dbc6b23e604b5c0d5ab322a494e772

### PR DESCRIPTION
**Pull request by XuHuaiyu**
fix #6221
PTAL @zhexuany @zz-jason

### CODE REVIEW

**Comment by zz-jason** [comment]
UT failed:
```
FAIL: builtin_time_test.go:1414: testEvaluatorSuite.TestUnixTimestamp

builtin_time_test.go:1495:
    c.Assert(str, Equals, test.expect, Commentf("%+v", test))
... obtained string = "1447410019.012"
... expected string = "1447410019.012000"
... {inputDecimal:0 input:{k:5 collation:0 decimal:0 length:0 i:0 b:[50 48 49 53 45 49 49 45 49 51 32 49 48 58 50 48 58 49 57 46 48 49 50] x:<nil>} expectKind:8 expect:1447410019.012000}
```

**Comment by jackysp** [comment]
Do we need to add the test case from #6221 ?

File: expression/builtin_time.go [hunk 1]
```diff
@@ -2897,6 +2897,20 @@ func (c *unixTimestampFunctionClass) getFunction(ctx sessionctx.Context, args []
 		if argEvaltp == types.ETString {
 			// Treat types.ETString as unspecified decimal.
 			retDecimal = types.UnspecifiedLength
+			if cnst, ok := args[0].(*Constant); ok {
+				tmpStr, isNull, err := cnst.EvalString(ctx, nil)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				if isNull {
```
Comment (zimulala) [inline expression/builtin_time.go]: Could we remove this judgment? I think this case is equal to `err != nil`.

Comment (zimulala) [reply expression/builtin_time.go]: Can we remove line 2946? I think it checked in line 2931.

**Comment by XuHuaiyu** [comment]
comments are addressed 
PTAL @jackysp @zimulala

**Pull request merged**